clients/pkg/promtail/targets/file/filetarget.go

```diff
@@ -196,8 +199,26 @@ func (t *FileTarget) sync() error {
 		}
 	}
 
+	if fi, err := os.Stat(t.pathExclude); err == nil && !fi.IsDir() {
+		matchesExcluded = []string{t.pathExclude}
+	} else {
+		matchesExcluded, err = doublestar.Glob(t.pathExclude)
```

Comment:
If path.Exclude is empty, then doublestar.Glob returns `nil, nil`, just skipping the loop below.